src/Database/Type/DateTimeType.php

```diff
@@ -200,6 +218,27 @@ public function toPHP($value, DriverInterface $driver)
         return $instance;
     }
 
+    /**
+     * Set whether DateTime object created from database string is converted
+     * to default time zone.
+     *
+     * If your database date times are in a specific time zone that you want
+     * to keep in the DateTime instance then set this to true.
+     *
+     * When false, datetime timezones are converted to default time zone.
+     * This is default behavior.
+     *
+     * @param bool $keep If true, database time zone is kept when converting
+     *      to DateTime instances.
+     * @return $this
+     */
+    public function setKeepDatabaseTimezone(bool $keep)
+    {
+        $this->keepDatabaseTimezone = $keep;
```

Comment:
What do non-timezone aware type mappers do when this is set to true? Should the timezone type be the class that implements this instead?

Reply:
They do the same thing.  Here's the most complex example:

1. MySQL is configured for 'Europe/London'
2. PHP is configured for 'America/Toronto'
3. `DateTimeType::setDatabaseTimezone()` is set to 'Europe/London'
4. `DateTimeType::toPHP()` converts a value from the database

With the default of `keepDatabaseTimezone=false` `toPHP` will return the `Time` instance converted to 'America/London'.  With `true`, it will leave the `Time` instance as-is in 'Europe/London'.

Reply:
This also allows time zone conversion to happen on the DB-side for databases that support time zone conversion that return a time zone in the string.  You'll get the instance in the time zone specified by the string.